packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -24,7 +55,52 @@ export class Snapshot {
     // TODO: CDX-347 Display Expanded preview
   }
 
-  async delete() {
-    // TODO: CDX-359: Delete snapshot once previewed
+  private get lastestReport(): ResourceSnapshotsReportModel {
+    if (this.model.reports === undefined) {
+      throw new Error(
+        `No detailed report found for the snapshot ${this.snapshotId}`
+      );
+    }
+    return this.model.reports.slice(-1)[0];
+  }
+
+  private isValid() {
+    const {resultCode, status} = this.lastestReport;
+    return (
+      status === ResourceSnapshotsReportStatus.Completed &&
+      resultCode === ResourceSnapshotsReportResultCode.Success
+    );
+  }
+
+  private async refreshSnapshotData() {
+    this.model = await this.snapshotClient.get(this.model.id, {
+      includeReports: true,
+    });
+  }
+
+  private async isIdle() {
+    await this.refreshSnapshotData();
+    return new Promise<void>((resolve) => {
```

Comment:
I'm not sure I understand why this is wrapped in a Promise ?

There's no rejection that would trigger the `waitUntilIdle` backoff mechanism, so... 🤔

Reply:
My bad. I forgot the `reject()`. I am not sure that is a good usage of the `exponential-backoff` package (as opposed to async-retry).